### DISCUSSION TITLE
Is there a vision or guide for discrete states and enumerations?

### DISCUSSION THREAD

**Discussion by hicksjacobp** [qudt/qudt-public-repo#478] [Q&A]
I see `qudt:DiscreteState` is defined in `schema/SCHEMA_QUDT-DATATYPE-v2.1.ttl`, but I see there are only 3 sub-classes: `qudt:OnOffStateType`, `qudt:OpenCloseStateType`, and `qudt:WetDryStateType`.

I then see that these 3 sub-classes go on to further define possible values of those states, for example:

```
qudt:On
  a qudt:OnOffStateType ;
  qudt:code "1" ;
  qudt:literal "on" ;
  rdfs:isDefinedBy <http://qudt.org/2.1/schema/datatype> ;
  rdfs:label "On" ;
.
```

However, I see that:

- `qudt:code` is marked as deprecated in `schema/SCHEMA_QUDT-v2.1.ttl`
- `qudt:literal` has a sub-class of `qudt:symbol`, but I don't understand the difference of when to use one over the other and the `qudt:On` example uses `qudt:literal` instead of `qudt:symbol`. `qudt:EnumeratedValue` has a constraint to be a sub-class of a `qudt:symbol`, and not `qudt:literal`.

I also see `qudt:Enumeration` in `schema/SCHEMA_QUDT-DATATYPE-v2.1.ttl` which I assume would allow one to define a set of discrete states that belong together.

I see that `qudt:DiscreteState` is a sub-class of `qudt:EnumeratedValue`, which has a few more examples not considered discrete states and also aren't considered enumerations.

And I can't seem to find much written guidance around enumerations throughout the QUDT website or GitHub repository. So I'm just looking for some insight and possible direction of where this ontology is taking enumerated values / states.

**Comment by steveraysteveray** [comment]
Speaking just for myself, this is a part of the QUDT ontology that has not been updated in years. Our focus has been on Quantifiable values, i.e. those that have a numerical value, rather than enumerations. As you correctly point out, the notion of enumerations is present in the ontology, but I haven't taken a look at that part of the model and am not surprised if there are some inconsistencies.
I don't pretend to speak for my colleagues at QUDT.org, though!

**Reply by PeteHazelberg** [reply]
Thanks @steveraysteveray for taking the time to weigh in with your perspective. That's helpful information.
